internal/otelmatchers/matchers.go

```diff
@@ -0,0 +1,84 @@
+package otelmatchers
+
+import (
+	"fmt"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func ConsistOfSpansWithIDs(expectedSpanIDs []string) types.GomegaMatcher {
+	return gomega.WithTransform(func(actual interface{}) ([]string, error) {
+		actualBytes, ok := actual.([]byte)
+		if !ok {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs rqquires a []byte, but got %T", actual)
+		}
+
+		actualSpans, err := unmarshalOTLPJSONTraceData(actualBytes)
+		if err != nil {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs requires a valid OTLP JSON document: %v", err)
+		}
+
+		var actualSpanIDs []string
+		for _, span := range actualSpans {
+			actualSpanIDs = append(actualSpanIDs, span.SpanID)
+		}
+		return actualSpanIDs, nil
+	}, gomega.ConsistOf(expectedSpanIDs))
+}
+
+func EachHaveTraceID(expectedTraceID string) types.GomegaMatcher {
+	return gomega.WithTransform(func(actual interface{}) ([]string, error) {
+		actualBytes, ok := actual.([]byte)
+		if !ok {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs rqquires a []byte, but got %T", actual)
```

Comment:
is this the wrong error message? it names `ConsistOfSpansWithIDs` in the `EachHaveTraceID` test

Reply:
Fixed